homeassistant/components/http.py

```diff
@@ -221,28 +220,6 @@ def send(self, request, filepath):
 _GZIP_FILE_SENDER = GzipFileSender()
 
 
-class HAStaticRoute(StaticRoute):
-    """StaticRoute with support for fingerprinting."""
-
-    def __init__(self, prefix, path):
-        """Initialize a static route with gzip and cache busting support."""
-        super().__init__(None, prefix, path)
-        self._file_sender = _GZIP_FILE_SENDER
-
-    def match(self, path):
-        """Match path to filename."""
-        if not path.startswith(self._prefix):
-            return None
-
-        # Extra sauce to remove fingerprinted resource names
```

Comment:
Frontend is experiencing the same behaviour on closer inspection, it appears that the tests suffer from a race when starting components, `test_frontend` is an example of this. Under a debugger I can see that `http` and `frontend` are started on separate threads, whilst HA is running on it's own loop thread - as expected. This means that `HomeAssistant` has significant chance (and typically always does) to fire it's started event *before* all components have completed their setup. 

I'm not sure how to proceed with this.